test/functional/mempool_accept.py

```diff
@@ -72,85 +69,71 @@ def run_test(self):
         assert_raises_rpc_error(-22, 'TX decode failed', lambda: node.testmempoolaccept(rawtxs=['ff00baar']))
 
         self.log.info('A transaction already in the blockchain')
-        coin = coins.pop()  # Pick a random coin(base) to spend
-        raw_tx_in_block = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': coin['txid'], 'vout': coin['vout']}],
-            outputs=[{node.getnewaddress(): 0.3}, {node.getnewaddress(): 49}],
-        ))['hex']
-        txid_in_block = node.sendrawtransaction(hexstring=raw_tx_in_block, maxfeerate=0)
+        raw_tx_in_block = miniwallet.send_self_transfer(from_node=node)
         node.generate(1)
         self.mempool_size = 0
         self.check_mempool_result(
-            result_expected=[{'txid': txid_in_block, 'allowed': False, 'reject-reason': 'txn-already-known'}],
-            rawtxs=[raw_tx_in_block],
+            result_expected=[{'txid': raw_tx_in_block['txid'], 'allowed': False, 'reject-reason': 'txn-already-known'}],
+            rawtxs=[raw_tx_in_block['hex']],
         )
 
         self.log.info('A transaction not in the mempool')
-        fee = Decimal('0.000007')
-        raw_tx_0 = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{"txid": txid_in_block, "vout": 0, "sequence": BIP125_SEQUENCE_NUMBER}],  # RBF is used later
-            outputs=[{node.getnewaddress(): Decimal('0.3') - fee}],
-        ))['hex']
-        tx = tx_from_hex(raw_tx_0)
+        fee = Decimal("0.003")  # The default fee_rate in MiniWallet
+        coin = miniwallet.get_utxo()  # Get a utxo and store it for later
+        raw_tx_0 = miniwallet.create_self_transfer(fee_rate=fee, from_node=node, utxo_to_spend=coin, sequence=BIP125_SEQUENCE_NUMBER) # RBF is used later
+        tx = tx_from_hex(raw_tx_0['hex'])
         txid_0 = tx.rehash()
```

Comment:
nit: I think the txid can simpy be obtained by accessing the dict field `txid` of the self-transfer method's result (rehashing shouldn't be needed here).

Reply:
Removed multiple instances where rehashing the transaction was not required. `tx.rehash()` is now used only when a `CTransaction()` object is changed.